File: desert/thunderhooves.lua
loop
    local yak = self:findNearest(self:findEnemies())
    if not yak then
        self:moveXY(self.pos.x+10, self.pos.y)
    elseif yak.pos.y > self.pos.y then
        self:buildXY("fence", yak.pos.x, yak.pos.y-10)
    else
        self:buildXY("fence", yak.pos.x, yak.pos.y+10)
    end
end

File: desert/thunderhooves.clj
(while true
  (let [yak (.findNearestEnemy this)]
    (if yak
        (if (> (.y (.pos yak)) (.y (.pos this)))
          (.buildXY this "fence" (.x (.pos yak)) 35)
          (.buildXY this "fence" (.x (.pos yak)) 25))
        (.move this (clj->js { :x 68, :y 30 })))))
